utils/release/push_to_artifactory

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import argparse
+import logging
+import os
+import re
+import requests
+import time
+
+from artifactory import ArtifactorySaaSPath
+
+
+# Necessary ENV variables
+def getenv(name, default=None):
+    env = os.getenv(name, default)
+    if env is not None:
+        return env
+    raise KeyError("necessary {} environment is not set".format(name))
+
+
+TEMP_PATH = getenv("TEMP_PATH", ".")
+# One of the following ENVs is necessary
+JFROG_API_KEY = getenv("JFROG_API_KEY", "")
+JFROG_TOKEN = getenv("JFROG_TOKEN", "")
+
+
+class Packages(object):
+    rpm_arch = dict(all="noarch", amd64="x86_64")
+    packages = (
+        ("clickhouse-client", "all"),
+        ("clickhouse-common-static", "amd64"),
+        ("clickhouse-common-static-dbg", "amd64"),
+        ("clickhouse-server", "all"),
+        ("clickhouse-test", "all"),
+    )
+
+    def __init__(self, version: str):
+        self.deb = tuple(
+            "_".join((name, version, arch + ".deb")) for name, arch in self.packages
+        )
+
+        rev = "2"
+        self.rpm = tuple(
+            "-".join((name, version, rev + "." + self.rpm_arch[arch] + ".rpm"))
+            for name, arch in self.packages
+        )
+
+    def arch(self, deb_pkg: str) -> str:
+        if not deb_pkg in self.deb:
+            raise ValueError("{} not in {}".format(deb_pkg, self.deb))
+        return deb_pkg.removesuffix(".deb").split("_")[-1]
+
+    @staticmethod
+    def path(package):
+        return os.path.join(TEMP_PATH, package)
+
+
+class S3(object):
+    template = (
+        "https://s3.amazonaws.com/"
+        # "clickhouse-builds/"
+        "{bucket_name}/"
+        # "33333/" or "0/"
+        "{pr}/"
+        # "2bef313f75e4cacc6ea2ef2133e8849ecf0385ec/"
+        "{commit}/"
+        # "clickhouse_build_check_(actions)/"
+        "{check_name}/"
+        # "clang-13_relwithdebuginfo_memory_bundled_unsplitted_notidy_without_coverage_deb/"
+        "{build_name}/"
+        # "clickhouse-common-static_21.11.5.0_amd64.deb"
+        "{package}"
+    )
+
+    def __init__(
+        self,
+        bucket_name: str,
+        pr: int,
+        commit: str,
+        check_name: str,
+        build_name: str,
+        version: str,
+    ):
+        self._common = dict(
+            bucket_name=bucket_name,
+            pr=pr,
+            commit=commit,
+            check_name=check_name,
+            build_name=build_name,
+        )
+        self.packages = Packages(version)
+
+    @staticmethod
+    def get_with_retries(url) -> requests.Response:
```

Comment:
BTW, I have this function https://github.com/ClickHouse/ClickHouse/blob/27ff9d1fc71beb94332e431c5ed9e9123e31d73d/tests/ci/build_download_helper.py#L28-L64, maybe it can be useful here.